subsys/bluetooth/host/Kconfig

```diff
@@ -510,8 +510,8 @@ config BT_RFCOMM_L2CAP_MTU
 	default BT_RX_BUF_LEN
 	default BT_L2CAP_RX_MTU if BT_HCI_ACL_FLOW_CONTROL
 	depends on BT_RFCOMM
-	range BT_RX_BUF_LEN 32767
 	range BT_L2CAP_RX_MTU 32767 if BT_HCI_ACL_FLOW_CONTROL
+	range BT_RX_BUF_LEN 32767
```

Comment:
is "range" different from "default", isnt the last evaluation that will be picked? with this change the range will always become BT_RX_BUF_LEN to 32767.

Reply:
@cvinayak 
Yep, it's different, which is confusing. The commit message in https://github.com/zephyrproject-rtos/zephyr/pull/9059/commits/f572976f2c094852af958a6c23a00a2b137793a1 goes over why.

That's one of the reasons that I'm trying to go back to standard Kconfig behavior for defaults. I don't know how many other similar mistakes there are.

Reply:
See http://docs.zephyrproject.org/porting/board_porting.html#zephyr-specific-kconfig-behavior-for-defaults as well. Only Zephyr does that prefer-later-defaults thing.

That it's inconsistent makes it even nastier. I'm planning to just turn all the defaults around later (along with having an easily-disabled warning about the changed behavior). There's been other gotchas too (for example, `choice`s don't prefer later defaults either).